frontend/ui/detailview.jsx

```diff
@@ -433,33 +454,67 @@ class DetailViewComponent extends React.Component {
             <Row>
               <form className="form-inline">
                 <select
-                  value={
-                    this.state.validTimeIntervals.findIndex(timeInterval =>
-                      timeInterval.interval === this.state.timeInterval &&
-                      ((this.state.startTime &&
-                        (timeInterval.startTime === this.state.startTime)) ||
-                       (!this.state.startTime && !timeInterval.startTime)))
-                  }
-                  onChange={this.timeIntervalChanged}
-                  className="mb-2 mr-sm-2 mb-sm-0">
-                  {
-                    this.state.validTimeIntervals
-                      .map((timeInterval, index) => (
-                        <option
-                          key={`${timeInterval.startTime || ''}-${timeInterval.interval}`}
-                          value={index} >
-                          {timeInterval.label}
-                        </option>
-                      ))
-                  }
-                  <option value="-1">Custom...</option>
+                  onChange={this.relativeChanged}
+                  className="mb-2 mr-sm-2 mb-sm-0"
```

Comment:
Gotta love the meaningful class names 😍 (JK)

Reply:
lol, yes, we should clean this up at some point...